### ISSUE TITLE
How can I determine or estimate when Aspect starts and ends

### ISSUE THREAD

**Issue by jeferson-sb** [astrorigin/pyswisseph#60]
Hi folks,

I'm working on implementing a Transits chart and I need to list aspects for planets in a particular day.
I'm able to determine the aspect between two bodies with swe_calc_ut successfully but I cannot estimate how long that aspect will occur. So I need to represent the two dates where that aspect will stay, for example

```
Sun opposition Mars  ✔️
20 Oct - 01 Nov
```

I know there is the `next_aspect_with` method under the contrib module but it can only display the future dates of that same aspect.

**Comment by jeferson-sb** [comment]
bump

**Comment by NodBr** [comment]
I may have an idea using to loops. For sake of simplicity, I will call:
- D+0 is the day you're interested in
- D+1, D+2, etc are the following days
- D-1, D-2, etc are the previous days. 

Start date:
- Create a new chart with the date D-1;
- Check If the aspect is still there;
- If it's not, you know it started today;
- If it's there, check D-2, D-3, etc

End date:
- Do the same but going forward;
- D+1, D+2, D+3, etc.
- When it's not there anymore, you know it ended

Depending on your project, after knowing the date, you may go deeper to find the hour using the same idea

**Comment by jeferson-sb** [comment]
@NodBr Yeah, I did something similar to this in our project, thanks for the response anyways.
